ext/phar/tar.c

```diff
@@ -607,7 +611,7 @@ int phar_parse_tarfile(php_stream* fp, char *fname, size_t fname_len, char *alia
 			phar_destroy_phar_data(myphar);
 			return FAILURE;
 		}
-	} while (!php_stream_eof(fp));
```

Comment:
Isn't it a problem then that the eof test has been removed? Here now 261: `pos = php_stream_tell(fp);` `pos` is unchecked  for error returns (are there any?; was unchecked before, too).

Probably can be moved to now 258 `while (true) {`  as `while (!php_stream_eof(fp)) {` ? @nielsdos

Reply:
`php_stream_tell(fp)` cannot fail.
`while (!php_stream_eof(fp)) {` won't work because the flag is not set in time, leading to the same issue as before.

Reply:
@nielsdos Thank you for the clarification.